src/library/export/exportrequest.h

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include <QSet>
+
+#include "library/trackset/crate/crateid.h"
+
+namespace mixxx {
+
+/// The ExportRequest struct is the abstract base class for all types of request
+/// to export the Mixxx library to some external location.  An export to a
+/// specific kind of external location is likely to need additional information
+/// to be specified in the request, which can be achieved by sub-classing this
+/// struct.
+struct ExportRequest {
+    virtual ~ExportRequest() = default;
+
+    /// Indicates whether to export only selected crates (if set to true), or
+    /// whether to export the whole Mixxx library (if set to false).
+    bool exportSelectedCrates;
```

Comment:
Using a base class without any behavior and only inherit data is an anti-pattern. It also doesn't work as intended, because it is sent *by-value*, i.e. due to slicing additional data from derived classes would be stripped off.

Please declare this struct as `final` and remove the virtual destructor.

Reply:
This base class is actually a hang-over from earlier designs: on reflection, I don't feel it serves any purpose to be separated, and I can simplify by removing it and just having `EnginePrimeExportRequest`.  I'll make this change.

Reply:
YAGNI strikes again ;)

Reply:
Now removed.